onnxruntime/python/onnxruntime_pybind_state_common.h

```diff
@@ -16,8 +15,36 @@ namespace python {
 using namespace onnxruntime;
 using namespace onnxruntime::logging;
 
-inline const SessionOptions& GetDefaultCPUSessionOptions() {
-  static SessionOptions so;
+struct CustomOpLibrary {
+  CustomOpLibrary(const char* library_path, OrtSessionOptions& ort_so);
+
+  ~CustomOpLibrary();
+
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(CustomOpLibrary);
+
+ private:
+  void* library_handle_ = nullptr;
+};
+
+struct CustomOpLibraries {
+  CustomOpLibraries() = default;
+
+  void AddLibrary(std::unique_ptr<CustomOpLibrary> custom_op_library);
+
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(CustomOpLibraries);
+
+ private:
+  std::vector<std::unique_ptr<CustomOpLibrary>> custom_op_libraries_;
```

Comment:
CustomOpLibraries will "own" the Customlibrary. The session options that needs to load the shared library needn't hold onto it after it has gotten what it needs from the library.